cmd/promctl/main.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/util/cli"
+	"github.com/prometheus/prometheus/version"
+)
+
+func check(t cli.Term, filename string) error {
+	t.Infof(fmt.Sprintf("Checking %s", filename))
+
+	if stat, err := os.Stat(filename); err != nil {
+		return fmt.Errorf("cannot get file info")
+	} else if stat.IsDir() {
+		return fmt.Errorf("is a directory")
+	}
+
+	switch ext := filepath.Ext(filename); strings.ToLower(ext) {
+	case ".yml", ".yaml":
+		t.Infof("  configuration file detected")
+
+		content, err := ioutil.ReadFile(filename)
+		if err != nil {
+			return err
+		}
+
+		var cfg config.Config
+		if err := yaml.Unmarshal(content, &cfg); err != nil {
+			return err
+		}
+		t.Infof("  SUCCESS")
+
+	case ".rules":
+		t.Infof("  rule file detected")
+
+		content, err := ioutil.ReadFile(filename)
+		if err != nil {
+			return err
+		}
+
+		rules, err := promql.ParseStmts(string(content))
+		if err != nil {
+			return err
+		}
+		t.Infof(fmt.Sprintf("  SUCCESS: %d rules found", len(rules)))
+
+	default:
+		return fmt.Errorf("Unknown file extension %q. (allowed = .yml, .yaml, .rules)", ext)
```

Comment:
Yes, we should. We are not doing it in other places for backwards compatibility – but we probably should. Can you think of a single good reason why one would choose a different extension?

It's this kind of strictness which keeps Go's eco system clean and makes the `go` tool just work.
We should strive for the same.

I'm strongly against introducing two instead of one command because 1 in 5000 users is not willing to do a `mv`.